pkg/util/limiter/utilization.go

```diff
@@ -144,6 +148,10 @@ func (l *UtilizationBasedLimiter) compute(now time.Time) (currCPUUtil float64, c
 		cpuUtil := (cpuTime - prevCPUTime) / now.Sub(prevUpdate).Seconds()
 		l.cpuMovingAvg.Add(int64(cpuUtil * 100))
 		l.cpuMovingAvg.Tick()
+		for i := 1; i < len(l.cpuSamples); i++ {
```

Comment:
An alternative might be to make `cpuSamples` a circular buffer and track where the head/tail are, so that we don't need to move all the samples on each `compute`. Logging the samples would need to start from the head and loop around the buffer.

Reply:
I implemented a circular buffer in the meantime.